cmd/block-generator/generator/generate.go

```diff
@@ -13,23 +14,62 @@ import (
 	"github.com/algorand/indexer/types"
 )
 
+const (
+	consensusTime int64 = 4500
+)
+
 // GenerationConfig defines the tunable parameters for block generation.
 type GenerationConfig struct {
-	// Block generation features
-	TxnPerBlock         uint64
-	NewAccountFrequency uint64
-
-	Protocol                     string
-	NumGenesisAccounts           uint64
-	GenesisAccountInitialBalance uint64
-	GenesisID                    string
-	GenesisHash                  sdk_types.Digest
+	Name                         string `mapstructure:"name"`
+	NumGenesisAccounts           uint64 `mapstructure:"genesis_accounts"`
+	GenesisAccountInitialBalance uint64 `mapstructure:"genesis_account_balance"`
+
+	// Block generation
+	TxnPerBlock uint64 `mapstructure:"tx_per_block"`
+
+	// TX Distribution
+	PaymentTransactionFraction float32 `mapstructure:"tx_pay_fraction"`
+	AssetTransactionFraction   float32 `mapstructure:"tx_asset_fraction"`
+
+	// Payment configuration
+	PaymentNewAccountFraction float32 `mapstructure:"pay_acct_create_fraction"`
+	PaymentTransferFraction   float32 `mapstructure:"pay_xfer_fraction"`
+
+	// Asset configuration
+	AssetCreateFraction  float32 `mapstructure:"asset_create_fraction"`
+	AssetDestroyFraction float32 `mapstructure:"asset_destroy_fraction"`
+	AssetOptinFraction   float32 `mapstructure:"asset_optin_fraction"`
+	AssetCloseFraction   float32 `mapstructure:"asset_close_fraction"`
+	AssetXferFraction    float32 `mapstructure:"asset_xfer_fraction"`
+}
+
+func sumIsCloseToOne(numbers ...float32) bool {
```

Comment:
Looks like most of the config fractions aren't used yet.
The other pattern I've seen is to just accept arbitrary float64 weights, sum them up, and then scale a random [0..1.0) to that sum and see where it lands in the lined up weights and pick that action.

Reply:
Yup, that's next. I realized that this was a fairly complete piece of functionality, so decided to make the PR.